daemon/workit.cpp

```diff
@@ -246,21 +242,33 @@ int work_it(CompileJob &j, unsigned int job_stat[], MsgChannel *client, CompileR
         }
 
         bool hasPipe = false;
+        bool hasSaveTemps = false;
 
+        // we could ignore this, except the silly gcc-4.8 thing with tthe caret fails the tests due
+        // to having -pipe and -save-temps=obj
+        for (std::list<string>::const_iterator it = list.begin();
```

Comment:
why didn't you use auto here?

Reply:
just copied the line below it ...
was sort of last minute, I didn't have this fix in my original since I don't use 4.8 normally but it happened to be on the box I was using

Reply:
problem was the test failed since it got some stderr and the 4.8 thing
but didn't actually error (even with -Werror), normally for my setup it just kept going ...
gcc: warning: -pipe ignored because -save-temps specified
I was looking at the fix in my setup to just clean it up but figured its so rarely used ... I didn't care (till the test failed)